core/lib/workarea/ext/freedom_patches/dragonfly_job_fetch_url.rb

```diff
@@ -15,7 +34,16 @@ def get(url)
           request.basic_auth(url.user, url.password)
         end
 
-        http.request(request)
+        response = http.request(request)
+
+        raise ProxyError, url if blocked?(response)
+
+        response
+      end
+
+      def blocked?(response)
```

Comment:
talking with @MMartyn, the proxy should throw back a 403 at us and this special header rather than time out. a timeout would be the result of _not_ having a proxy configured at all, which shouldn't be the case here so long as you're using our hosting.